lib/taxonomy/taxon.rb

```diff
@@ -0,0 +1,58 @@
+module Taxonomy
+  class Taxon
+    extend Forwardable
+    attr_reader :name, :content_id, :base_path
+    attr_accessor :parent_node, :children
+    def_delegators :tree, :map, :each
+
+    def initialize(title:, base_path:, content_id:, **_args)
+      @name = title
+      @content_id = content_id
+      @base_path = base_path
+      @children = []
+    end
+
+    def tree
+      return [self] if @children.empty?
```

Comment:
FWIW this class is also copy-pasta, and I plan on improving it later. Didn't want to rewrite the whole world in this first PR.